src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -35,6 +36,11 @@ protected string PathToManagedToolWithoutExtension
         /// </summary>
         protected abstract string PathToNativeTool { get; }
 
+        protected ManagedToolTask(ResourceManager resourceManager)
+            : base(resourceManager)
```

Comment:
Assigning `TaskResources` doesn't actually fully initialize the resource lookup of the `Task`. The only way to actually set it up correctly is to pass it through the constructor. All our `Task` derive from this hence this is a good choke point to add enforcement.